### PR TITLE
Implement new claim API and do minor updates

### LINKED ISSUE #296: Specify API for claims nonces and versions
Currently the approach is to give the nonce and version in the claim constructor.  Usually, the user will not care about the nonce and version, and it will be the Issuer class the one in charge of deciding a nonce and the version.  It would be nice to have a user API for building claims that doesn't require specifying nonce and version.

### PR DESCRIPTION
**Pull request by ed255**
Resolve #296

The new Claim API works with a new Metadata struct that allows setting the
following properties of a claim in a generic way:
- Optional Recipient Identity
- Optional Version
- Optional Expiration date
- Revocation Nonce

Every Claim must contain the metadata object.  The new claims.EntrierMetadata
interface allows setting metadata properties from functions that consume claims
in a generic way.  A claim header must be defined for every claim type to
specify the flags that the claim type uses.  For now only ClaimBasic and
ClaimAuthorizeKSignBabyJub have been defined.

Other updates:
- Remove idenmanager (deprecated)
- Remove idenadminutils (deprecated)
- Remove deprecated functions from `core/proof`
- Add helper script to generate godoc links
- Update README with godoc links

### CODE REVIEW

File: core/claims/claim.go [hunk 1]
```diff
@@ -131,35 +111,187 @@ func NewClaimFromEntry(e *merkletree.Entry) (merkletree.Entrier, error) {
...
+	case ClaimRecipIdenValue:
+		copy(value[1][:], m.Dest[:])
+	default:
+		panic(fmt.Sprintf("Unexpected header.Dest %v", m.header.Dest))
+	}
+	if m.header.Version {
+		binary.BigEndian.PutUint32(index[0][ClaimTypeLen+ClaimFlagsLen:], m.Version)
+	}
+	if m.header.Expiration {
+		binary.BigEndian.PutUint64(value[0][32/8:], uint64(m.Expiration))
```
Comment (arnaucube) [inline core/claims/claim.go]: Why ClaimExpirationLen is a hardcoded value, while ClaimTypeLen,ClaimVersionLen, ClaimFlagsLen, ClaimFlagsLen, ClaimHeaderLen, ClaimVersionLen, ClaimRevNonceLen are declared as constants? In this way when we need to have a claim that needs Expiration, we have the constant of the len to do the parser.
With the ClaimExpirationLen or not, maybe here should use `value[0][ClaimRevNonceLen:]`?

File: core/claims/claim.go [hunk 2]
```diff
@@ -131,35 +111,187 @@ func NewClaimFromEntry(e *merkletree.Entry) (merkletree.Entrier, error) {
...
+	ClaimHeaderAuthorizeKSignBabyJub = ClaimHeader{
+		Type:       *ClaimTypeAuthorizeKSignBabyJub,
+		Dest:       ClaimRecipSelf,
+		Expiration: false,
+		Version:    false}
+)
+
+// EntrierMetadata is an intefrace that extends Entrier with a function that
+// returns the claim metadata.
+type EntrierMetadata interface {
```
Comment (arnaucube) [inline core/claims/claim.go]: Proposal: `EntrierMetadata` --> `Claimer`.
The idea is that a `Claimer` interface is what all the `Claims` are satisfying. So the interface don't get a name like `EntrierMetadata`, when is a superset of `Entrier`, and is more natural as superset as the `Claimer` contain a `Entrier`.

File: core/claims/claim.go [hunk 3]
```diff
@@ -131,35 +111,187 @@ func NewClaimFromEntry(e *merkletree.Entry) (merkletree.Entrier, error) {
...
+	case ClaimRecipIdenValue:
+		copy(m.Dest[:], value[1][:])
+	default:
+		panic(fmt.Sprintf("Unexpected header.Dest %v", m.header.Dest))
+	}
+	if m.header.Version {
+		m.Version = binary.BigEndian.Uint32(index[0][ClaimTypeLen+ClaimFlagsLen:])
+	}
+	if m.header.Expiration {
+		m.Expiration = int64(binary.BigEndian.Uint64(value[0][32/8:]))
```
Comment (arnaucube) [inline core/claims/claim.go]: Same as previous comment

File: core/claims/claimBasic.go [hunk 4]
```diff
@@ -1,85 +1,70 @@
...
-	"encoding/binary"
-
 	"github.com/iden3/go-iden3-core/merkletree"
 )
 
 const (
-	// 56+248+248+248=800 bits
-	indexSlotBits  = 800
+	// 152+248+248+248=896 bits
+	indexSlotBits  = 896
```
Comment (arnaucube) [inline core/claims/claimBasic.go]: this shouldn't be 120+248+248+248= 864?

File: core/claims/claimBasic.go [hunk 5]
```diff
@@ -1,85 +1,70 @@
...
 	IndexSlotBytes = indexSlotBits / 8
 	// 216+248+248+248=960 bits
-	dataSlotBits  = 960
-	DataSlotBytes = dataSlotBits / 8
+	valueSlotBits  = 960
```
Comment (arnaucube) [inline core/claims/claimBasic.go]: this shouldn't be 152+248+248+248=896?

Basically we moved 64 bits from Index to Value, so Index was 800 bits and Value was 960, moving those 64 bits from I to V, results with 864 bits for Index, and 896 bits for Value, witch matches with the numbers calculated in the review comments.
See the Claims specification docs for more details

File: core/claims/claim.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ed255) [reply core/claims/claim.go]: I like this proposal!

File: core/claims/claimBasic.go (hunk 4 above)
Comment (ed255) [reply core/claims/claimBasic.go]: I used 152 because the version field is not being used in this claim.

File: core/claims/claimBasic.go (hunk 5 above)
Comment (ed255) [reply core/claims/claimBasic.go]: I used 216 because the expiration date is not being used in this claim.

File: core/claims/claimBasic.go (hunk 4 above)
Comment (arnaucube) [reply core/claims/claimBasic.go]: Right, perfect

File: core/claims/claimBasic.go (hunk 5 above)
Comment (arnaucube) [reply core/claims/claimBasic.go]: Right, perfect

File: core/claims/claim.go (hunk 3 above)
Comment (ed255) [reply core/claims/claim.go]: I have modified the code to use the `ClaimRevNonceLen` constant

File: core/claims/claim.go (hunk 1 above)
Comment (ed255) [reply core/claims/claim.go]: I have modified the code to use the `ClaimRevNonceLen` constant

File: core/claims/claim.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ed255) [reply core/claims/claim.go]: Done!

**Comment by ed255** [comment]
I have addressed the requested changes.  I've also changed the declaration of `IndexSlotBytes` and `ValueSlotLen` to be more explicit about how they are being calculated.  The values remain the same.

**Pull request merged**